File: 88-MergeSortedArray/88-MergeSortedArray.java
// Last updated: 16/10/2025, 04:29:45
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
       
        int ans[] = new int[m+n];
        
        
        if(n != 0){ //if n == 0 then no need to change anything
          int i = 0;
          int j = 0;
          int k = 0;
          while(i<m && j<n){
            if(nums1[i] <= nums2[j]){
                ans[k++] = nums1[i++];
            }else{
                ans[k++] = nums2[j++];
            }
          }
            
          if(i == m){
                while(k<m+n){
                    ans[k++] = nums2[j++];
                }
            }
            else if(j == n){
                while(k<m+n){
                    ans[k++] = nums1[i++];
                }
            }
            for(i=0;i<ans.length;i++){
              nums1[i] = ans[i];
            }
        }
    }
}

File: 88-MergeSortedArray/88-MergeSortedArray.cpp
// Last updated: 16/10/2025, 04:29:53



        
    


class Solution {

public:

    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {

        //Agar samajh nahi aaye toh jara gaur farmao dono array sorted hai :)

        int p=m-1,q=n-1,i=m+n-1;

        while(q>=0){

            if(p>=0 && nums1[p]>nums2[q]){

                nums1[i--]=nums1[p--];

            }

            else{

                nums1[i--]=nums2[q--];

            }

        }

    }

};